### DISCUSSION TITLE
Neovim pug support?

### DISCUSSION THREAD

**Discussion by bjorntrondsen** [vuejs/language-tools#4494] [General]
Is it possible to add `@vue/language-plugin-pug` to a Neovim lsp setup? I'm mainly looking to add syntax highlighting to `<template lang="pug">` tags. Been trying a few different ways but it is not working. My config:

```lua
local mason_registry = require('mason-registry')
local vue_language_server_path = mason_registry.get_package('vue-language-server'):get_install_path() .. '/node_modules/@vue/language-server'
local typescript_path = mason_registry.get_package('vue-language-server'):get_install_path() .. '/node_modules/typescript/lib'
lspconfig.tsserver.setup({
  init_options = {
    plugins = {
      {
        name = '@vue/typescript-plugin',
        location = vue_language_server_path,
        languages = { 'vue' },
      },
      {
        name = '@vue/language-plugin-pug',
        location = vue_language_server_path,
        languages = { 'vue' },
      },
    },
  },
  filetypes = { 'typescript', 'javascript', 'javascriptreact', 'typescriptreact', 'vue' },
})
lspconfig.volar.setup({
  init_options = {
    typescript = {
      tsdk = typescript_path
    },
  }
})
```

**Comment by so1ve** [comment]
Syntax highlighting should be handled by neovim itself, not volar, right?

**Comment by bjorntrondsen** [comment]
I was using vim-vue (+ vim-pug before), but it's not maintained.
The [readme](https://github.com/posva/vim-vue) says ` This project is currently not actively maintained. We recommend to check the official vuejs/language-tools instead`, which is why I was looking here.

I'm not very familiar with how things work under the hood but I guess LSPs can do processing to provide semantic tokens, which neovim then uses for the syntax highlighting

**Comment by RayGuo-ergou** [comment]
This lsp does not handle all highlighting e.g. css, so you have to install separately. Not sure what are you using to handle syntax highlighting but if you are using treesitter, you can install it via `TSInstall pug`.
